### DISCUSSION TITLE
Quarkus support for arm64

### DISCUSSION THREAD

**Discussion by jvanzyl** [quarkusio/quarkus#27553] [Community]
Hi,

I see a bunch of discussions in various GitHub issues, stackoverflow threads, at Zulip streams but what is the current state of being able to build `arm64` binaries and images?

**Comment by cescoffier** [comment]
We are actively working on providing a way to build ARM64 executable.
This is a two-steps epic:

1. provide a way to build ARM64 executables and AMR64 container images from an ARM machine (M1, M2, Graviton...)
2. provide a way to build multi-arch images and cross-compile native executables


1) is in progress and should land very soon. The images are already available on https://quay.io/user/cescoffi, so you can test them by overriding the container image to do the build and replace the base image of your docker file. 

2) is going to take more time. Our current approach uses QEMU (so virtualization), but we are facing unacceptable build time (it is not "go drink a coffee" but more "go have lunch with friends in a 5 stars restaurant, and maybe it will be complete when you come back). So, we will rework our approach as soon as 1. lands in "main".

**Reply by jvanzyl** [reply]
Fantastic, thanks for the update!
